tests/algorithm_tests.cpp

```diff
@@ -0,0 +1,66 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+// Copyright (c) 2015 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include <UnitTest++/UnitTest++.h>
+#include <gsl/gsl_algorithm>
+
+#include <array>
+
+using namespace std;
+using namespace gsl;
+
+SUITE(algorithm_tests)
+{
+
+    TEST(copy_span_dynamic_same_type)
+    {
+        std::array<int, 5> src{ 1,2,3,4,5 };
+        std::array<int, 10> dst{};
+
+        span<int> src_span(src);
+        span<int> dst_span(dst);
+
+        copy(src_span, dst_span);
+
+        copy(src_span, dst_span.subspan(src_span.size()));
+
+        for (std::ptrdiff_t i = 0; i < static_cast<std::ptrdiff_t>(src.size()); ++i) {
+            CHECK(dst_span[i] == src[i]);
+            CHECK(dst_span[i + src.size()] == src[i]);
+        }
+    }
+
+    TEST(copy_span_static_diff_type)
```

Comment:
This is for a type that *is* assignable. Could we have a test that validates that the reverse assignment (for example) does not compile? Similarly, we should probably validate that you can't accidentally slice objects using this function.